File: packages/js-dpp/lib/stateTransition/fee/calculateStateTransitionFeeFactory.js
/**
 * @param {calculateStateTransitionFeeFromOperations} calculateStateTransitionFeeFromOperations
 * @returns {calculateStateTransitionFee}
 */
function calculateStateTransitionFeeFactory(calculateStateTransitionFeeFromOperations) {
  /**
   * @typedef {calculateStateTransitionFee}
   * @param {AbstractStateTransition} stateTransition
   * @return {{
   *   storageFee: number,
   *   processingFee: number,
   *   feeRefunds: {identifier: Buffer, creditsPerEpoch: Object<string, number>}[],
   *   totalRefunds: number,
   *   total: number,
   *   desiredAmount: number,
   *   requiredAmount: number,
   * }}
   */
  function calculateStateTransitionFee(stateTransition) {
    const executionContext = stateTransition.getExecutionContext();

    return calculateStateTransitionFeeFromOperations(
      executionContext.getOperations(),
      stateTransition.getOwnerId(),
    );
  }

  return calculateStateTransitionFee;
}

module.exports = calculateStateTransitionFeeFactory;

File: packages/rs-dpp/src/state_transition/fee/calculate_state_transition_fee_factory.rs
use crate::state_transition::state_transition_execution_context::StateTransitionExecutionContext;
use crate::state_transition::{
    fee::calculate_state_transition_fee_from_operations_factory::calculate_state_transition_fee_from_operations,
    StateTransition, StateTransitionLike,
};

use super::FeeResult;

pub fn calculate_state_transition_fee(
    state_transition: &StateTransition,
    execution_context: &StateTransitionExecutionContext,
) -> FeeResult {
    calculate_state_transition_fee_from_operations(
        &execution_context.get_operations(),
        state_transition.get_owner_id(),
    )
}
